assets/js/monero-gateway-order-page.js

```diff
@@ -0,0 +1,161 @@
+    jQuery('#monero_total_amount').html(details.amount_total_formatted);
+    jQuery('#monero_total_paid').html(details.amount_paid_formatted);
+    jQuery('#monero_total_due').html(details.amount_due_formatted);
+
+    jQuery('#monero_integrated_address').html(details.integrated_address);
+
+    if(monero_show_qr) {
+        var qr = jQuery('#monero_qr_code').html('');
+        new QRCode(qr.get(0), details.qrcode_uri);
+    }
+
+    if(details.txs.length) {
+        jQuery('#monero_tx_table').show();
+        jQuery('#monero_tx_none').hide();
+        jQuery('#monero_tx_table tbody').html('');
+        for(var i=0; i < details.txs.length; i++) {
+            var tx = details.txs[i];
+            var height = tx.height == 0 ? 'N/A' : tx.height;
+            var row = ''+
+                '<tr>'+
+                '<td style="word-break: break-all">'+
+                '<a href="'+monero_explorer_url+'/tx/'+tx.txid+'" target="_blank">'+tx.txid+'</a>'+
+                '</td>'+
+                '<td>'+height+'</td>'+
+                '<td>'+tx.amount_formatted+' Monero</td>'+
+                '</tr>';
+
+            jQuery('#monero_tx_table tbody').append(row);
+        }
+    } else {
+        jQuery('#monero_tx_table').hide();
+        jQuery('#monero_tx_none').show();
+    }
+
+    // Show state change notifications
+    var new_txs = details.txs;
+    var old_txs = monero_order_state.txs;
+    if(new_txs.length != old_txs.length) {
+        for(var i = 0; i < new_txs.length; i++) {
+            var is_new_tx = true;
+            for(var j = 0; j < old_txs.length; j++) {
+                if(new_txs[i].txid == old_txs[j].txid && new_txs[i].amount == old_txs[j].amount) {
+                    is_new_tx = false;
+                    break;
+                }
+            }
+            if(is_new_tx) {
+                monero_showNotification('Transaction received for '+new_txs[i].amount_formatted+' Monero');
+            }
+        }
+    }
+
+    if(details.status != monero_order_state.status) {
+        switch(details.status) {
+            case 'paid':
+                monero_showNotification('Your order has been paid in full');
+                break;
+            case 'confirmed':
+                monero_showNotification('Your order has been confirmed');
+                break;
+            case 'expired':
+            case 'expired_partial':
+                monero_showNotification('Your order has expired', 'error');
+                break;
+        }
+    }
+
+    monero_order_state = {
+        status: monero_details.status,
+        txs: monero_details.txs
+    };
+
+}
+jQuery(document).ready(function($) {
+    if (typeof monero_details !== 'undefined') {
+        monero_order_state = {
+            status: monero_details.status,
+            txs: monero_details.txs
+        };
+        setInterval(monero_fetchDetails, 30000);
+        monero_updateDetails();
+        new ClipboardJS('.clipboard').on('success', function(e) {
```

Comment:
I do not know why you need to add some random JS library for "Clipboarding".

Reply:
Because it's not standard across browsers, and mobile platforms, which is why clipboard.js exists in the first place. Originally I did not use a library and it wasn't cross-platform.

Reply:
Why do you have to copy a string on mobile when you can click on a link monero:// which will open a Monero wallet (Monerujo and Cakewallet supports monero:// )?

Reply:
If this is a serious objection, then clipboarding can be done the way the masari plugin does it, however clipboard.js isn't a heavy framework, but I understand if you want to drop it.